File: packages/core/types/template/DataTypes.ts
import { IBaseNode, TemplateNodeTypes } from "./BaseNode/BaseTypes";

export interface IEnumType extends IBaseNode {
  name: string;
  nodeType: TemplateNodeTypes.EnumType;
}

export interface IStructType extends IBaseNode {
  name: string;
  nodeType: TemplateNodeTypes.StructType;
}

export interface ITypeDefinition extends IBaseNode {
  nodeType: TemplateNodeTypes.TypeDefinition;
  name: string;
  underlyingType: string;
}

export interface IUnionType extends IBaseNode {
  nodeType: TemplateNodeTypes.UnionType;
  name: string;
}

File: packages/core/src/core/types/template/data_types.py
"""Data type node definitions."""

from .BaseNode.base_types import IBaseNode, TemplateNodeTypes


class IEnumType(IBaseNode):
    """Enum type node."""

    name: str
    nodeType: TemplateNodeTypes = TemplateNodeTypes.EnumType


class IStructType(IBaseNode):
    """Struct type node."""

    name: str
    nodeType: TemplateNodeTypes = TemplateNodeTypes.StructType


class ITypeDefinition(IBaseNode):
    """Type definition node."""

    nodeType: TemplateNodeTypes = TemplateNodeTypes.TypeDefinition
    name: str
    underlyingType: str


class IUnionType(IBaseNode):
    """Union type node."""

    nodeType: TemplateNodeTypes = TemplateNodeTypes.UnionType
    name: str


